packages/react/src/examples/forms/validationStatic.stories.tsx

```diff
@@ -304,15 +328,18 @@ export const Static = () => {
             </div>
             {formik.values.parkingPeriod === 'temporary' && (
               <div className="hds-example-form__item">
-                <TextInput
+                <DateInput
                   id="permitEndDate"
                   name="permitEndDate"
                   label="Permit end date"
                   helperText="Use format DD.MM.YYYY"
-                  onChange={formik.handleChange}
+                  onChange={(value) => {
```

Comment:
I noticed that the form validates the value when the date picker is opened. I think it should not show error messages when the user is about to select a date. I'm not sure if there is a way to prevent this.

Reply:
This seemed to happen if you first go to the input field, and click the date picker button after that. If you directly click date picker, then there's no error. I guess there's a simple validate on blur. And the onBlur gets called if you click the date picker button when input is focused.

Reply:
Ok, so it might work correctly then

Reply:
Hmm.. maybe the onBlur should not happen if we open the datepicker? Not sure if it's possible to do it so that it does not happen

Reply:
But I guess it is DateInput issue? 🤔

Reply:
Yes, it might be actually something about our TextInput component, that is applied by DateInput. It seems the onBlur is only passed to the input element, and not to the whole wrapper. That would explain why the onBlur is called, even the focus still stays for the whole wrapper. Not sure if there's any way fixing it without causing a breaking chance. Also as the TextInput is used in so many other components, it might be scary to change its behaviour. To fix the issue we would have to remove the onBlur from the input, and then set it to the input wrapper instead.

Reply:
As suggested by a11y audit, this can be dealt with by calling the onBlur validation only if the field is dirty:

- [ ] Validate onBlur only if the field is dirty